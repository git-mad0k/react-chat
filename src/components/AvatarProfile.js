import React from 'react'
import Avatar from '@material-ui/core/Avatar'
import titleInitilals from '../utils/title-initials'
import getColor from '../utils/color-from'


const AvatarProfile = ({ name, color }) => (
  <Avatar style={{ backgroundColor: color ? color: getColor(name)}}>
    {titleInitilals(name)}
  </Avatar>)

export default AvatarProfile
