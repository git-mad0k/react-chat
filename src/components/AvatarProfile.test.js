/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AvatarProfile from './AvatarProfile';

describe('<AvatarProfile />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AvatarProfile name="Name Surname" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('отображается корректно', () => {
    const tree = renderer.create(<AvatarProfile name="Name Surname" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
