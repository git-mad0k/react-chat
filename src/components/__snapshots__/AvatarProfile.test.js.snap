// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarProfile /> отображается корректно 1`] = `
<div
  className="MuiAvatar-root-4 MuiAvatar-colorDefault-5"
  style={
    Object {
      "backgroundColor": "#2196f3",
    }
  }
>
  NS
</div>
`;
